lightning-block-sync/src/init.rs

```diff
@@ -26,6 +26,47 @@ BlockSourceResult<ValidatedBlockHeader> where B::Target: BlockSource {
 		.validate(best_block_hash)
 }
 
+/// Polls the block source, returning a validated block header and [`BestBlock`]
+/// corresponding to the block source's best chain tip.
+///
+/// Upon success, the returned header can be used to initialize [`SpvClient`],
+/// and the returned [`BestBlock`] can be passed into [`ChannelManager::new`]
+/// as part of its [`ChainParameters`]. Initializing the channel manager and spv
+/// client with the outputs of this function ensures that the channel manager
+/// and [`SpvClient`] are synced to the same block during a fresh start.
+///
+/// [`SpvClient`]: crate::SpvClient
+/// [`ChainParameters`]: lightning::ln::channelmanager::ChainParameters
+/// [`ChannelManager::new`]: lightning::ln::channelmanager::ChannelManager::new
+// This prevents a possible race where an intermediate block is mined in between
+// when the channel manager and SpvClient are initialized, causing them to be
+// out of sync. When the SpvClient detects the next block, it feeds this new
+// block into the channel manager, but the channel manager is actually expecting
+// to hear about the intermediate block, thereby panicking with the message
+// "Blocks must be connected in chain-order - the connected header must build on
+// the last connected header."
+pub async fn validate_best_block_header_and_best_block<B: Deref>(block_source: B)
+-> BlockSourceResult<(ValidatedBlockHeader, BestBlock)>
+where
+	B::Target: BlockSource
+{
+	let (polled_best_block_hash, maybe_best_block_height) = block_source
+		.get_best_block()
+		.await?;
+	let best_block_header_data = block_source
+		.get_header(&polled_best_block_hash, maybe_best_block_height)
+		.await?;
+	let best_block_height = maybe_best_block_height.ok_or_else(||
+			BlockSourceError::persistent("block height missing")
+		)?;
```

Comment:
`BlockHeaderData` includes the height, so no need to fail. This height is just a hint for the `BlockSource` when fetching the header.

Reply:
Removed this function entirely as per your suggestion